### PR TITLE
[action] Added `format` option to `swiftlint` action

### PR DESCRIPTION
**Pull request by bguidolim**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

This PR adds the option `format` to `swiftlint` action.

### Description

The option is only available for mode `autocorrect`, it formats the code besides the changes regarding the rules. If the mode is set to `lint`, then the option has no effect.

### CODE REVIEW

File: fastlane/lib/fastlane/actions/swiftlint.rb [hunk 1]
```diff
@@ -114,7 +115,12 @@ def self.available_options
           FastlaneCore::ConfigItem.new(key: :executable,
                                        description: "Path to the `swiftlint` executable on your machine",
                                        is_string: true,
-                                       optional: true)
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :format,
+                                        description: "Format code when mode is :autocorrect",
```
Comment (giginet) [inline fastlane/lib/fastlane/actions/swiftlint.rb]: `indent` 👀

File: fastlane/lib/fastlane/actions/swiftlint.rb [hunk 2]
```diff
@@ -114,7 +115,12 @@ def self.available_options
...
+                                        default_value: false,
```
Comment (giginet) [inline fastlane/lib/fastlane/actions/swiftlint.rb]: You can specify `type: Boolean,`.

fastlane has `Boolean` type.

File: fastlane/spec/actions_specs/swiftlint_spec.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 describe Fastlane do
   describe Fastlane::FastFile do
     describe "SwiftLint" do
-      let(:swiftlint_gem_version) { Gem::Version.new('0.9.2') }
+      let(:swiftlint_gem_version) { Gem::Version.new('0.11.0') }
```
Comment (giginet) [inline fastlane/spec/actions_specs/swiftlint_spec.rb]: Should you change this value?
This change may affect to exist tests.

I think it is better to override this value only for new test cases.

File: fastlane/spec/actions_specs/swiftlint_spec.rb [hunk 4]
```diff
@@ -289,6 +289,56 @@
...
+            swiftlint(
+              mode: :lint,
+              format: true
+            )
+          end").runner.execute(:test)
+
+          expect(result).to eq("swiftlint lint")
+        end
+
+        it "omits the switch if swiftlint version is too low" do
```
Comment (giginet) [inline fastlane/spec/actions_specs/swiftlint_spec.rb]: too earlier? (I'm not a native speaker)

File: fastlane/spec/actions_specs/swiftlint_spec.rb [hunk 5]
```diff
@@ -289,6 +289,56 @@
...
+          allow(Fastlane::Actions::SwiftlintAction).to receive(:swiftlint_version).and_return(Gem::Version.new('0.10.0'))
+
+          result = Fastlane::FastFile.new.parse("lane :test do
+            swiftlint(
+              mode: :autocorrect,
+              format: true
+            )
+          end").runner.execute(:test)
+
```
Comment (giginet) [inline fastlane/spec/actions_specs/swiftlint_spec.rb]: You're better to test whether ` UI.important` is called with the message.

**Review by giginet — COMMENTED** [review]
Good job!

I pointed a few. Please check it out

File: fastlane/spec/actions_specs/swiftlint_spec.rb (hunk 3 above)
Comment (bguidolim) [reply fastlane/spec/actions_specs/swiftlint_spec.rb]: I think so, by default I believe we should use the most recent version, but for tests, I set the latest version comparing with the other options. If I don't change it here, then I need to set the version for every test I made.

File: fastlane/spec/actions_specs/swiftlint_spec.rb (hunk 4 above)
Comment (bguidolim) [reply fastlane/spec/actions_specs/swiftlint_spec.rb]: It's not right, but I basically copied and pasted. I'm gonna change this.

File: fastlane/spec/actions_specs/swiftlint_spec.rb (hunk 5 above)
Comment (bguidolim) [reply fastlane/spec/actions_specs/swiftlint_spec.rb]: I didn't get your point.

**Review by joshdholtz — APPROVED** [review]
@bguidolim This is 💯! Thanks for the contribution ❤️

**Pull request merged**